### PR TITLE
feat: initial chat UI merge

### PR DESCRIPTION
**Pull request by cpacker**
Refactored version of #474

---

To use:

1. (`cd chatui`) `pnpm install` - Install deps via npm
2. (`cd chatui`) `pnpm dev` - Start the chat UI web server (should default to http://localhost:3000/)
3. `memgpt server` - In a separate tab/terminal start the backend server

---

### Starting server from `memgpt`
```sh
% memgpt server
INFO:     Started server process [57128]
INFO:     Waiting for application startup.
Writing out openapi.json file
INFO:     Application startup complete.
INFO:     Uvicorn running on http://localhost:8283 (Press CTRL+C to quit)
```

### Starting frontend from `memgpt/chatui`
```sh
chatui % npx nx serve
> nx run memgpt:serve:development

  ➜  Local:   http://localhost:4200/
```

### Browser at `http://localhost:4200`:
image

### CODE REVIEW

File: memgpt/config.py [hunk 1]
```diff
@@ -219,7 +219,7 @@ def __post_init__(self):
 
     @staticmethod
     def generate_uuid() -> str:
-        return uuid.UUID(int=uuid.getnode()).hex
+        return str(uuid.UUID(int=uuid.getnode()))
```
Comment (cpacker) [inline memgpt/config.py:222]: ?

File: memgpt/client/client.py [hunk 2]
```diff
@@ -67,7 +67,7 @@ def list_agents(self):
 
     def agent_exists(self, agent_id: str) -> bool:
         existing = self.list_agents()
-        return agent_id in existing["agent_names"]
+        return agent_id in [agent["name"] for agent in existing["agents"]]
```
Comment (cpacker) [inline memgpt/client/client.py:70]: Possible source of test failures?

**Review by sarahwooders — APPROVED** [review]
I only looked over the parts in the existing codebase, but lgtm!

**Review by sarahwooders — APPROVED** [review]
lgtm!

**Pull request merged**